File: packages/features/feature_test/lib/src/test/components/test_header.dart
import 'package:flutter/material.dart';
import 'package:hooks_riverpod/hooks_riverpod.dart';
import '../test_notifier.dart';

class TestHeader extends ConsumerWidget {
  const TestHeader({super.key});

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final state = ref.watch(testProvider);
    final theme = Theme.of(context);

    return Column(
      mainAxisSize: MainAxisSize.min,
      children: [
        Padding(
          padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
          child: Row(
            mainAxisAlignment: MainAxisAlignment.spaceBetween,
            children: [
              // Back Button
              SizedBox(
                width: 40,
                height: 40,
                child: IconButton(
                  onPressed: () => _confirmExit(context, ref),
                  icon: Icon(
                    Icons.arrow_back,
                    size: 22,
                    color: theme.colorScheme.onBackground,
                  ),
                ),
              ),

              // Timer
              if (state.timeLimitSeconds > 0)
                _TimerDisplay(
                  seconds: state.timeRemainingSeconds,
                  isUrgent: state.timeRemainingSeconds < 60,
                ),

              // Favorite Button (Placeholder logic for now)
              SizedBox(
                width: 40,
                height: 40,
                child: IconButton(
                  onPressed: () {
                    // TODO: Implement toggle favorite
                  },
                  icon: const Icon(
                    Icons.favorite_border,
                    size: 24,
                    // Note: TestDanger red would be used if favorite
                  ),
                  color: theme.colorScheme.onBackground,
                ),
              ),
            ],
          ),
        ),
        Divider(
          height: 1, 
          thickness: 1, 
          color: theme.colorScheme.surfaceVariant,
        ),
      ],
    );
  }

  void _confirmExit(BuildContext context, WidgetRef ref) {
    showDialog(
      context: context,
      builder: (context) => AlertDialog(
        title: const Text("退出测试"),
        content: const Text("确定要放弃本次测试吗？进度将不会保留。"),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context),
            child: const Text("继续测试"),
          ),
          TextButton(
            onPressed: () {
              Navigator.pop(context); // Close dialog
              Navigator.pop(context); // Exit test screen
            },
            child: Text(
              "退出",
              style: TextStyle(color: Theme.of(context).colorScheme.error),
            ),
          ),
        ],
      ),
    );
  }
}

class _TimerDisplay extends StatelessWidget {
  final int seconds;
  final bool isUrgent;

  const _TimerDisplay({required this.seconds, required this.isUrgent});

  @override
  Widget build(BuildContext context) {
    final minutes = seconds ~/ 60;
    final remainingSeconds = seconds % 60;
    final timeStr = 
        "${minutes.toString().padLeft(2, '0')}:${remainingSeconds.toString().padLeft(2, '0')}";

    return Padding(
      padding: const EdgeInsets.symmetric(horizontal: 8),
      child: Text(
        timeStr,
        style: TextStyle(
          fontFamily: 'Monospace',
          fontWeight: FontWeight.bold,
          color: isUrgent ? Colors.red : Theme.of(context).colorScheme.onBackground,
          fontSize: 16,
        ),
      ),
    );
  }
}

File: archive/feature/test/src/main/java/com/jian/nemo/feature/test/components/TestHeader.kt
package com.jian.nemo.feature.test.components

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FavoriteBorder
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.jian.nemo.core.domain.model.Word
import com.jian.nemo.feature.test.presentation.theme.TestDanger

/**
 * 测试头部组件（包含返回按钮、倒计时和收藏按钮）
 * 复刻旧项目 com.jian.nemo.ui.screen.test.components.TestHeader
 */
import com.jian.nemo.core.domain.model.Grammar

@Composable
fun TestHeader(
    onBack: () -> Unit,
    timeLimitSeconds: Int,
    timeRemainingSeconds: Int,
    word: Word?,
    grammar: Grammar? = null,
    onToggleFavorite: (Int, Boolean) -> Unit
) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(horizontal = 16.dp, vertical = 12.dp),
        horizontalArrangement = Arrangement.SpaceBetween,
        verticalAlignment = Alignment.CenterVertically
    ) {
        // 返回按钮
        IconButton(
            onClick = onBack,
            modifier = Modifier.size(40.dp)
        ) {
            Icon(
                Icons.AutoMirrored.Filled.ArrowBack,
                contentDescription = "返回菜单",
                tint = MaterialTheme.colorScheme.onBackground,
                modifier = Modifier.size(22.dp)
            )
        }

        // 添加倒计时显示
        if (timeLimitSeconds > 0) {
            val minutes = timeRemainingSeconds / 60
            val seconds = timeRemainingSeconds % 60
            val isRunningOut = timeRemainingSeconds < 60
            Text(
                text = "%02d:%02d".format(minutes, seconds),
                color = if (isRunningOut) TestDanger else MaterialTheme.colorScheme.onBackground,
                fontWeight = FontWeight.Bold,
                modifier = Modifier.padding(horizontal = 8.dp),
                fontFamily = FontFamily.Monospace
            )
        }

        // 收藏按钮
        // 只有当有单词信息时才显示收藏按钮（例如非打字题或已加载）
        // 收藏按钮
        // 只有当有单词或语法信息时才显示收藏按钮
        val isFavorite = word?.isFavorite == true || grammar?.isFavorite == true
        val itemId = word?.id ?: grammar?.id ?: 0

        if (itemId != 0) {
            IconButton(
                onClick = { onToggleFavorite(itemId, !isFavorite) },
                modifier = Modifier.size(40.dp)
            ) {
                Icon(
                    imageVector = if (isFavorite) Icons.Default.Favorite else Icons.Default.FavoriteBorder,
                    contentDescription = "收藏",
                    tint = if (isFavorite) TestDanger else MaterialTheme.colorScheme.onBackground,
                    modifier = Modifier.size(24.dp)
                )
            }
        }
    }

    // 添加底部边框
    HorizontalDivider(color = MaterialTheme.colorScheme.surfaceVariant, thickness = 1.dp)
}
